src/daft-local-execution/src/resource_manager.rs

```diff
@@ -0,0 +1,116 @@
+use std::sync::{Arc, Mutex};
+
+use common_error::{DaftError, DaftResult};
+use common_system_info::SystemInfoInternal;
+use lazy_static::lazy_static;
+use tokio::sync::Notify;
+
+lazy_static! {
+    pub static ref MEMORY_MANAGER: Arc<MemoryManager> = Arc::new(MemoryManager::new());
+}
+
+fn custom_memory_limit() -> Option<u64> {
+    let memory_limit_var_name = "DAFT_MEMORY_LIMIT";
+    if let Ok(val) = std::env::var(memory_limit_var_name) {
+        if let Ok(val) = val.parse::<u64>() {
+            return Some(val);
+        }
+    }
+    None
+}
+
+pub fn get_memory_manager() -> Arc<MemoryManager> {
```

Comment:
I would instead return a `Arc<MemoryManager>`, up to the consume to clone if they need it